repository/role_perm_postgres.go

```diff
@@ -0,0 +1,171 @@
+		logrus.Errorf("GetRoleById: error while scanning for role:%s", err)
+		return nil, fmt.Errorf("GetRoleById: repository error:%w", err)
+	}
+	return &role, transaction.Commit()
+}
+
+func (r *RolePermPostgres) GetAllRoles() ([]model.Role, error) {
+	transaction, err := r.db.Begin()
+	if err != nil {
+		logrus.Errorf("GetAllRoles: can not starts transaction:%s", err)
+		return nil, fmt.Errorf("GetAllRoles: can not starts transaction:%w", err)
+	}
+	var roles []model.Role
+	query := "SELECT id, name FROM roles"
+	rows, err := transaction.Query(query)
+	if err != nil {
+		logrus.Errorf("GetAllRoles: can not executes a query:%s", err)
+		return nil, fmt.Errorf("GetAllRoles: repository error:%w", err)
+	}
+	for rows.Next() {
+		var role model.Role
+		if err := rows.Scan(&role.ID, &role.Name); err != nil {
+			logrus.Errorf("GetAllRoles: error while scanning for role:%s", err)
+			return nil, fmt.Errorf("GetAllRoles: repository error:%w", err)
+		}
+		roles = append(roles, role)
+	}
+	return roles, transaction.Commit()
+}
+
+func (r *RolePermPostgres) CreateRole(role string) (int, error) {
+	transaction, err := r.db.Begin()
+	if err != nil {
+		r.logger.Errorf("CreateRole: can not starts transaction:%s", err)
+		return 0, fmt.Errorf("createRole: can not starts transaction:%w", err)
+	}
+	var roleId int
+	defer transaction.Rollback()
+	row := transaction.QueryRow("INSERT INTO roles (name) VALUES ($1) RETURNING id", role)
+	if err := row.Scan(&roleId); err != nil {
+		r.logger.Errorf("CreateRole: error while scanning for roleId:%s", err)
+		return 0, fmt.Errorf("createRole: error while scanning for roleId:%w", err)
+	}
+	return roleId, transaction.Commit()
+}
+
+func (r *RolePermPostgres) BindRoleWithPerms(rp *model.BindRoleWithPermission) error {
+	transaction, err := r.db.Begin()
+	if err != nil {
+		r.logger.Errorf("BindRoleWithPerms: can not starts transaction:%s", err)
+		return fmt.Errorf("BindRoleWithPerms: can not starts transaction:%w", err)
+	}
+	defer transaction.Rollback()
+	query := "INSERT INTO role_permissions (role_id, permission_id) VALUES ($1, $2)"
+	for _, perm := range rp.PermissionsId {
+		_, err := transaction.Exec(query, rp.RoleId, perm)
+		if err != nil {
+			r.logger.Errorf("BindRoleWithPerms: error while binding role:%d and permissin:%d:%s", rp.RoleId, perm, err)
+			return fmt.Errorf("BindRoleWithPerms: error while binding role:%d and permissin:%d:%w", rp.RoleId, perm, err)
+		}
+	}
+	return transaction.Commit()
+}
+
+func (r *RolePermPostgres) GetPermsByRoleId(id int) ([]model.Permission, error) {
+	transaction, err := r.db.Begin()
+	if err != nil {
+		logrus.Errorf("GetPermsByRoleId: can not starts transaction:%s", err)
+		return nil, fmt.Errorf("GetPermsByRoleId: can not starts transaction:%s", err)
+	}
+	var permissions []model.Permission
+	query := "SELECT id, description FROM permissions JOIN role_permissions ON permissions.id = role_permissions.permission_id AND role_permissions.role_id = $1"
```

Comment:
when you have multiple tables better have table prefix in SELECT too. I mean `SELECT permissions.id` instead of `SELECT id`. Btw, you can also give table alias that would make the request shorter. `SELECT p.id FROM permissions p JOIN ....`